### ISSUE TITLE
SonarLint does not detect unused event handlers...

### ISSUE THREAD

**Issue by usmrtester** [SonarSource/sonar-csharp#857]
### Description

SonorLint does not detect events which are unused, see the following code.  Asdf() is detected as unused, but SaveButton_Click() is not.  Both have 0 references.

```
        0 references
        private void Asdf()
        {
            const int y = 10;

            for (int i = 0; i < y; i++)
            {
                Console.WriteLine("i = " + i.ToString());
            }
        }

        0 references
        private void SaveButton_Click(object sender, EventArgs e)
        {
            const int z = 10;

            for (int i = 0; i < z; i++)
            {
                Console.WriteLine("i = " + i.ToString());
            }
        }

```
### Repro steps

See above

### Expected behavior

Detect unused event handlers

0 references for unused event handlers, with no SonarLint detection

* SonarC# Version : 3.5.0.1803
* Visual Studio Version: 2017 version 15.3.5
* .Net Framework version 4.6.01590

**Comment by valhristov** [comment]
Hi @usmrtester, we are explicitly excluding methods that could possibly be event handlers because in WPF, for example, it is possible to add private event handler in XAML.
